core/src/main/java/org/apache/carbondata/core/scan/filter/executer/ExcludeFilterExecuterImpl.java

```diff
@@ -18,56 +18,152 @@
+      for (int i = 0; i < dimensionColumnDataChunks.length; i++) {
+        BitSet bitSet = getFilteredIndexes(dimensionColumnDataChunks[i],
+            dimensionRawColumnChunk.getRowCount()[i]);
+        bitSetGroup.setBitSet(bitSet, i);
+      }
+
+      return bitSetGroup;
+    } else if (isMeasurePresentInCurrentBlock == true) {
+      int blockIndex = segmentProperties.getMeasuresOrdinalToBlockMapping()
+          .get(msrColumnEvaluatorInfo.getColumnIndex());
+      if (null == blockChunkHolder.getMeasureRawDataChunk()[blockIndex]) {
+        blockChunkHolder.getMeasureRawDataChunk()[blockIndex] = blockChunkHolder.getDataBlock()
+            .getMeasureChunk(blockChunkHolder.getFileReader(), blockIndex);
+      }
+      MeasureRawColumnChunk measureRawColumnChunk =
+          blockChunkHolder.getMeasureRawDataChunk()[blockIndex];
+      MeasureColumnDataChunk[] measureColumnDataChunks =
+          measureRawColumnChunk.convertToMeasureColDataChunks();
+      BitSetGroup bitSetGroup = new BitSetGroup(measureRawColumnChunk.getPagesCount());
+      DataType msrType = getMeasureDataType(msrColumnEvaluatorInfo);
+      for (int i = 0; i < measureColumnDataChunks.length; i++) {
+        BitSet bitSet =
+            getFilteredIndexes(measureColumnDataChunks[i], measureRawColumnChunk.getRowCount()[i],
+                msrType);
+        bitSetGroup.setBitSet(bitSet, i);
+      }
+      return bitSetGroup;
     }
-    DimensionRawColumnChunk dimensionRawColumnChunk =
-        blockChunkHolder.getDimensionRawDataChunk()[blockIndex];
-    DimensionColumnDataChunk[] dimensionColumnDataChunks =
-        dimensionRawColumnChunk.convertToDimColDataChunks();
-    BitSetGroup bitSetGroup =
-        new BitSetGroup(dimensionRawColumnChunk.getPagesCount());
-    for (int i = 0; i < dimensionColumnDataChunks.length; i++) {
-      BitSet bitSet = getFilteredIndexes(dimensionColumnDataChunks[i],
-          dimensionRawColumnChunk.getRowCount()[i]);
-      bitSetGroup.setBitSet(bitSet, i);
+    return null;
+  }
+
+  private DataType getMeasureDataType(MeasureColumnResolvedFilterInfo msrColumnEvaluatorInfo) {
+    switch (msrColumnEvaluatorInfo.getType()) {
+      case SHORT:
+        return DataType.SHORT;
+      case INT:
+        return DataType.INT;
+      case LONG:
+        return DataType.LONG;
+      case DECIMAL:
+        return DataType.DECIMAL;
+      default:
+        return DataType.DOUBLE;
     }
+  }
 
-    return bitSetGroup;
+  protected BitSet getFilteredIndexes(MeasureColumnDataChunk measureColumnDataChunk,
+      int numerOfRows, DataType msrType) {
+    // Here the algorithm is
+    // Get the measure values from the chunk. compare sequentially with the
+    // the filter values. The one that matches sets it Bitset.
+    BitSet bitSet = new BitSet(numerOfRows);
+    bitSet.flip(0, numerOfRows);
+    byte[][] filterValues = msrColumnExecutorInfo.getFilterKeys();
```

Comment:
As of now using ByteArray in filterKeys, in later optimization will convert will store objects.